### ISSUE TITLE
DNS_Lexicon missing from Docker image

### ISSUE THREAD

**Issue by dschaper** [Neilpang/acme.sh#1148]
Docker image reports that lexicon is not installed.

Steps to reproduce
------------------
Run acme.sh docker image in daemon mode.

Debug log
-----------------

```sh
$ docker exec -it acme.sh sh
# lexicon
sh: lexicon: not found
# exit
$ which lexicon    
/usr/local/bin/lexicon
```

**Comment by dschaper** [comment]
Looking deeper at this, the dns_lexicon doesn't currently build a binary so adding the Python environment to the docker image would bloat it quite a bit. Might have to either look at building that into a binary in a build stage and make the image a multi-stage build or see if AnalogJ would be interested in building a binary release and freeze the Python code.

**Comment by dschaper** [comment]
I think I can build dns_lexicon in to a smaller binary using Docker multi-stage builds, then the python dependencies would only be in the build stage and the binaries could be copied over to the final running stage, reducing the image size tremendously. I'm in the middle of rewiriting my own TLS cert gen daemon in POSIX compliant shell to run inside an Alpine docker container, but I'll face the same issues with dns_lexicon, if I can crack the problem, I'll be sure to help and commit back a PR with my findings. I was planning on waiting for LE to do wildcards, but that looks to be passed back to March at the earliest. (I of course am attributing you and another project in all of my code to link back to my inspirations.)

**Comment by Neilpang** [comment]
I just added a new branch  `acme.sh:lexicon`

https://github.com/Neilpang/acme.sh/commit/8984178e03a20ef0088c7daa2e9c8aba33425a28


please try the docker image:  `neilpang/acme.sh:lexicon`

**Comment by Neilpang** [comment]
It increases the image size from 13MB  to 66MB.  It seems to expensive to me.
I doubt how many users need lexicon here.
So, I don't want to merge this branch into maser for now.  just use it as you want.

**Comment by dschaper** [comment]
I agree with the size bloat, when I can get a chance I can look at building the binary. That would be a multistage build so I think that limits the version under which you could build the dockerfile personally, but hub.docker.com supports multistage now so you could build an image to be pulled for the people that are running docker daemon < 17.05. Just need to freeze out the python and I think I can get it down to 4MB or so additional space.

But thanks for the branch, you can kill it off if you'd like since it's just a `pip` install run command.

**Comment by darkguy2008** [comment]
Thank for the lexicon branch, but it doesn't seem to work with wildcard names. Any ideas?
```
~/docker/acme$ sudo docker run --rm -it -e "PROVIDER=vultr" -e "LEXICON_VULTR_USERNAME=XXXXXXXX" -e "LEXICON_VULTR_TOKEN=XXXXXXXX" -v "$(pwd)/certs":/acme.sh --net=host neilpang/acme.sh:lexicon --issue -d xxxxx.com -d *.xxxxx.com --dns dns_lexicon
[Sun Nov 11 10:34:34 UTC 2018] It seems that *.xxxxx.com is an IDN( Internationalized Domain Names), please install 'idn' command first.
```

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Not a bug: bundling python/lexicon would bloat the Docker image, so it is intentionally not included in the default image.
